### PR TITLE
 SWARM-1946 +  SWARM-1948

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

**Comment by mkouba** [comment]
Oops, I forgot to rebase first...

**Comment by mkouba** [comment]
retest this please

File: fractions/microprofile/microprofile-fault-tolerance/src/main/java/org/wildfly/swarm/microprofile/faulttolerance/MicroProfileFaultToleranceFraction.java [hunk 1]
```diff
@@ -28,4 +31,12 @@
 
     public MicroProfileFaultToleranceFraction() {
     }
+
+    public boolean isSynchronousCircuitBreakerEnabled() {
+        return synchronousCircuitBreaker.get();
+    }
+
+    @AttributeDocumentation("Enable/disable synchronous circuit breaker functionality. If disabled, `CircuitBreaker#successThreshold()` of value greater than 1 is not supported. Moreover, circuit breaker does not necessarily transition from `CLOSED` to `OPEN` immediately when a fault tolerance operation completes. However, applications are encouraged to disable this feature on high-volume circuits.")
+    @Configurable("swarm.micro-profile-faulttolerance.synchronousCircuitBreaker")
```
Comment (Ladicek) [inline fractions/microprofile/microprofile-fault-tolerance/src/main/java/org/wildfly/swarm/microprofile/faulttolerance/MicroProfileFaultToleranceFraction.java:40]: Oh! We're crazy inconsistent in how MP fractions name their config keys :-(

Comment (mkouba) [reply fractions/microprofile/microprofile-fault-tolerance/src/main/java/org/wildfly/swarm/microprofile/faulttolerance/MicroProfileFaultToleranceFraction.java:40]: Yes, we do. Pls create a separate issue. Thanks.

File: fractions/netflix/hystrix/src/main/java/org/wildfly/swarm/netflix/hystrix/HystrixFraction.java [hunk 2]
```diff
@@ -74,6 +74,10 @@ public String streamPath() {
     @Configurable("swarm.hystrix.command.default.execution.isolation.semaphore.maxConcurrentRequests")
     private Defaultable<Integer> semaphorMaxConcurrentRequests = integer(10);
 
+    @AttributeDocumentation("The maximum number of requests allowed to a HystrixCommand.getFallback() method when you are using ExecutionIsolationStrategy.SEMAPHORE")
+    @Configurable("swarm.hystrix.command.default.fallback.isolation.semaphore.maxConcurrentRequests")
+    private Defaultable<Integer> fallbackSemaphoreMaxConcurrentRequests = integer(10);
```
Comment (Ladicek) [inline fractions/netflix/hystrix/src/main/java/org/wildfly/swarm/netflix/hystrix/HystrixFraction.java:79]: Is this even used somewhere, or is it added just for documentation purpose?

**Comment by Ladicek** [comment]
CC @tradej for docs review.

File: fractions/netflix/hystrix/src/main/java/org/wildfly/swarm/netflix/hystrix/HystrixFraction.java (hunk 2 above)
Comment (mkouba) [reply fractions/netflix/hystrix/src/main/java/org/wildfly/swarm/netflix/hystrix/HystrixFraction.java:79]: It is (but the original MP config property has higher priority): https://github.com/wildfly-swarm/wildfly-swarm/pull/872/files#diff-f7046bd48826f565b0ce20f18ff0e912R96

Comment (Ladicek) [reply fractions/netflix/hystrix/src/main/java/org/wildfly/swarm/netflix/hystrix/HystrixFraction.java:79]: That's a "is sync circuit breaker enabled", which is a different key.

Comment (Ladicek) [reply fractions/netflix/hystrix/src/main/java/org/wildfly/swarm/netflix/hystrix/HystrixFraction.java:79]: I remember now that we have a code somewhere that takes all `swarm.hystrix.*` (or something like that) and automatically pushes it to Hystrix. I think.

Comment (mkouba) [reply fractions/netflix/hystrix/src/main/java/org/wildfly/swarm/netflix/hystrix/HystrixFraction.java:79]: Ah, sorry. Yes, this one maps to https://github.com/Netflix/Hystrix/wiki/Configuration#fallback.isolation.semaphore.maxConcurrentRequests and was missing previously in `HystrixFraction`. Without this you wouldn't be able to configure the default value via YAML or system property. I don't know anything about the "code somewhere".

File: fractions/microprofile/microprofile-fault-tolerance/src/main/java/org/wildfly/swarm/microprofile/faulttolerance/MicroProfileFaultToleranceFraction.java (hunk 1 above)
Comment (Ladicek) [reply fractions/microprofile/microprofile-fault-tolerance/src/main/java/org/wildfly/swarm/microprofile/faulttolerance/MicroProfileFaultToleranceFraction.java:40]: Sure, filed https://issues.jboss.org/browse/SWARM-1963

**Comment by mkouba** [comment]
retest this please

**Comment by kenfinnigan** [comment]
retest this please

**Review by tradej — CHANGES_REQUESTED** [review]
Please hold the PR, I will submit docs fixes later today (I'm on mobile now). If it needs to go in earlier, it can (the docs are OK for the most part), but please file a follow-up issue then.

**Comment by tradej** [comment]
Changes submitted in mkouba#1.

**Review by tradej — APPROVED** [review]

**Review by Ladicek — APPROVED** [review]

**Pull request merged**